react-accordion-component/accordion/src/App.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState } from 'react';
+import './App.css';
+
+const topics = [
+  {
+    id: 1,
+    title: 'Hypertext Markup Language',
+    content:
+      'Hypertext Markup Language (HTML) is the standard ... for the World Wide Web',
+  },
+  {
+    id: 2,
+    title: 'Cascading Style Sheets',
+    content:
+      'Cascading Style Sheets (CSS) is a style sheet ... of the World Wide Web',
+  },
+  {
+    id: 3,
+    title: 'Javascript',
+    content:
+      'Javascript, often abbreviated as JS, is ... and first-class functions',
+  },
+];
+
+type Topic = {
+  id: number;
+  title: string;
+  content: string;
+};
+
+type AccordionProps = {
+  topics: Topic[];
+};
+
+function App() {
+  return <Accordion topics={topics} />;
+}
+
+const Accordion: React.FC<AccordionProps> = ({ topics }) => {
+  const [activeTopicId, setActiveTopicId] = useState<number | null>(null);
```

Comment:
If you use `undefined` as your "none" state instead of `null`, it gets a little easier since TypeScript figures things out for you. This works:
`const [activeTopicId, setActiveTopicId] = useState<number>();`

and automatically sets the type of `activeTopicId` to be `number | undefined`. Then you just use `undefined` instead of `null` in your code.

Since I personally believe that `null` was a mistake and should never have been added to the JS language, I much prefer this alternative approach.